herbert_core/classes/MPIFramework/messages/InitMessage.m

```diff
@@ -37,22 +37,16 @@
             %                 a cellarray it assumed to be 1
             %
             obj = obj@aMessage('init');
-            if ~exist('common_data', 'var')
-                common_data = [];
-                loop_data = 1;
-            end
-            if ~exist('loop_data', 'var')
-                loop_data = 1;
-            end
-            if ~exist('return_results', 'var')
-                return_results = false;
-            end
-            obj.payload = struct('common_data',common_data,...
-                'loopData',[],'n_first_step',1,'n_steps',0,...
-                'return_results',return_results );
-            if ~exist('n_first_step', 'var')
-                n_first_step = 1;
-            end
+            p = inputParser();
+            addOptional('common_data', [])
+            addOptional('loop_data', 1)
+            addOptional('return_results', false, @isboolean)
```

Comment:
I think this should be `islogical`, or you could use Herbert's `islognumscalar`, which permits `0` and `1` (doubles) as well as `true` and `false`